src/main/java/javax/jmdns/impl/ServiceInfoImpl.java

```diff
@@ -418,15 +418,35 @@ public String getHostAddress() {
      */
     @Override
     public String[] getHostAddresses() {
-        Inet4Address[] ip4Aaddresses = this.getInet4Addresses();
-        Inet6Address[] ip6Aaddresses = this.getInet6Addresses();
-        String[] names = new String[ip4Aaddresses.length + ip6Aaddresses.length];
-        for (int i = 0; i < ip4Aaddresses.length; i++) {
-            names[i] = ip4Aaddresses[i].getHostAddress();
+        Inet4Address[] ip4Addresses = this.getInet4Addresses();
+        Inet6Address[] ip6Addresses = this.getInet6Addresses();
+
+        // Default to empty arrays if null
+        if (ip4Addresses == null) {
+            ip4Addresses = new Inet4Address[0];
+        }
+        if (ip6Addresses == null) {
+            ip6Addresses = new Inet6Address[0];
         }
-        for (int i = 0; i < ip6Aaddresses.length; i++) {
-            names[i + ip4Aaddresses.length] = "[" + ip6Aaddresses[i].getHostAddress() + "]";
+
+        String[] names = new String[ip4Addresses.length + ip6Addresses.length];
+
+        for (int i = 0; i < ip4Addresses.length; i++) {
+            if (ip4Addresses[i] != null) {
+                names[i] = ip4Addresses[i].getHostAddress();
+            } else {
+                names[i] = "null"; // Handle null elements as needed
```

Comment:
The question here is what is "as needed"?
Filling the array with "null" strings looks dangerous to me - a caller of the method probably expects valid IP addresses in the array. I would hope that this else-branch will actually never be entered; but if so, I would rather log a message (maybe INFO level) and not add anything to the array (which will require changes to how the array is constructed and filled).
Wdyt?

Reply:
Sure thing i was trying to be as least impactful but I like your idea.

Reply:
Switched to INFO messages